### PR TITLE
Don.t share db handler object among APIs; retrieve single doc always …

### LINKED ISSUE #6764: request search intermitently give error 500
https://cmsweb.cern.ch/reqmgr2/data/request?status=assignment-approved&details=true

 is failing very often

### PR DESCRIPTION
**Pull request by amaltaro**
…when possible
Together with #6942, partially fixes #6764 

As spotted and suggested by Valentin, create a different db api object for each Rest API.
Second change is to avoid making a POST call to couchdb/_all_docs when the user is requesting a single doc.

The `500 Internal Server Error` is still showing up in the logs, though in a much smaller rate (< 15%)...
still investigating ...

@vkuznet please review
I still need to test it with the new tag in a clean environment

### CODE REVIEW

File: src/python/WMCore/Services/Requests.py [hunk 1]
```diff
@@ -236,7 +227,7 @@ def f():
 
         headers["Content-length"] = str(len(encoded_data))
 
-        assert isinstance(encoded_data, type('string')), \
+        assert isinstance(encoded_data, str), \
```
Comment (vkuznet) [inline src/python/WMCore/Services/Requests.py]: please always use basestring instead of str, it covers both use case a string and unicode.

**Comment by vkuznet** [comment]
Alan, changes are fine, but does not really solve the problem. For instance you may further improve your PR by converting makeRequest_httplib and  _getURLOpener methods https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/Requests.py#L358
to a stand-alone functions.

**Comment by vkuznet** [comment]
oops, here is link to makeRequest_httplib:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/Requests.py#L168

File: src/python/WMCore/Services/Requests.py (hunk 1 above)
Comment (amaltaro) [inline src/python/WMCore/Services/Requests.py]: From the previous check, I assumed we'd be interested only in 8-bit strings, not in unicode strings. Ok, I'll update the code soon

**Comment by amaltaro** [comment]
Giving it a try at pycurl...

**Comment by amaltaro** [comment]
several of them failed while locating data from PhEDEx.
I saw a couple others failing with http code 411, e.g.:
https://cmssdt.cern.ch/jenkins/job/DMWM-WMCore-PR-test/2492/testReport/junit/WMCore_t.RequestManager_t.ReqMgr_t/ReqMgrTest/testK_CheckRequestFailsInjectionForbiddenInputArg/

however, I do see `Content-Length` being provided in the headers, according to the same logs above.

I'll remove the last commit and keep only the initial changes proposed in this PR. Further changes will go in a new PR.

**Comment by amaltaro** [comment]
As discussed with Valentin, further improvements to make it thread safe will be done in a new PR (which will require much more testing), they are basically the split up of some of these methods unsafe into functions.

**Comment by vlimant** [comment]
Will this go in July deployment ?

**Comment by amaltaro** [comment]
Yes

**Pull request merged**